src/cedict-service/cedict.js

```diff
@@ -0,0 +1,389 @@
+   */
+  static checkConfiguration (configuration) {
+    if (!configuration.storage) throw new Error('Storage tree is missing from a configuration')
+    if (!configuration.storage.stores) throw new Error('Stores data is missing from a configuration')
+    if (!configuration.storage.stores.dictionary) throw new Error('Dictionary tree is missing from a configuration')
+    if (!configuration.storage.stores.dictionary.primaryIndex) throw new Error('A primaryIndex tree of a dictionary is missing from a configuration')
+    if (!configuration.storage.stores.dictionary.primaryIndex.hasOwnProperty('keyPath')) throw new Error('A keyPath option of a primaryIndex tree of a dictionary is missing from a configuration') // eslint-disable-line no-prototype-builtins
+    if (!configuration.storage.stores.dictionary.volatileStorage) throw new Error('A volatileStorage tree of a dictionary is missing from a configuration')
+    if (!configuration.storage.stores.dictionary.volatileStorage.hasOwnProperty('enabled')) throw new Error('enabled option of a volatileStorage tree of a dictionary is missing from a configuration') // eslint-disable-line no-prototype-builtins
+    if (!configuration.storage.stores.dictionary.volatileStorage.hasOwnProperty('indexed')) throw new Error('indexed option of a volatileStorage tree of a dictionary is missing from a configuration') // eslint-disable-line no-prototype-builtins
+    if (!configuration.storage.stores.dictionary.permanentStorage) throw new Error('A permanentStorage tree of a dictionary is missing from a configuration')
+    if (!configuration.storage.stores.dictionary.permanentStorage.hasOwnProperty('enabled')) throw new Error('enabled option of a permanentStorage tree of a dictionary is missing from a configuration') // eslint-disable-line no-prototype-builtins
+    if (!configuration.storage.stores.dictionary.permanentStorage.hasOwnProperty('indexed')) throw new Error('indexed option of a permanentStorage tree of a dictionary is missing from a configuration') // eslint-disable-line no-prototype-builtins
+    if (!configuration.data) throw new Error('Date tree is missing from a configuration')
+    if (!configuration.data.version) throw new Error('Data version is missing from a configuration')
+    if (!configuration.data.revision) throw new Error('Data revision is missing from a configuration')
+    if (!configuration.data.recordsCount) throw new Error('Data records count is missing from a configuration')
+    if (!configuration.data.URI) throw new Error('Data URI is missing from a configuration')
+    if (!configuration.data.chunks || configuration.data.chunks.length === 0) throw new Error('Data chunks are missing from a configuration')
+  }
+
+  /**
+   * Initializes a data object.
+   *
+   * @returns {Promise<undefined> | Promise<Error>} - A promise
+   */
+  init () {
+    return new Promise((resolve, reject) => {
+      this._storage = new CedictPermanentStorage(this._configuration.storage)
```

Comment:
so, right now, we always try to create the permanentStorage (i..e the indexedDb) even if the configuration is to use volatileStorage? Is this what we want to do?

Reply:
This is because we want to store data permanently to avoid downloading it with each page reload. So even it config specifies that data is in-memory only, we will still record it into IndexedDB and will try to read it from IndexedDB into memory upon the next page reload. Does it makes sense?

I will add more comments to clarify this point.

Reply:
I think that makes sense. I suppose there might be some scenario where we don't want to write anything at all to the indexedDb but unless there is a clear need defined for that I think it would probably be premature optimization.